### PR TITLE
Adding omg.lol DNS API

### LINKED ISSUE #4466: dnsapi request: omg.lol
Would be pretty handy: https://api.omg.lol/#dns

### PR DESCRIPTION
**Pull request by as-kholin**
DNS API for omg.lol.  

Fixes #4466 .

Bug Report issue: [https://github.com/acmesh-official/acme.sh/issues/5299](https://github.com/acmesh-official/acme.sh/issues/5299)
API Wiki Entry: [https://github.com/acmesh-official/acme.sh/wiki/dnsapi2#dns_omglol](https://github.com/acmesh-official/acme.sh/wiki/dnsapi2#dns_omglol)

### CODE REVIEW

**Comment by as-kholin** [comment]
Reviewed both API dev guide and API Test documentation, and believe I am following them consistently.

File: dnsapi/dns_omglol.sh [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+  fulldomain=$3
+
+  if [ "" = "$omglol_address" ]; then
+    _err "omg.lol base address not provided.  Exiting"
+    exit 1
+  fi
+
+  if [ "" = "$omglol_apikey" ]; then
+    _err "omg.lol API key not provided.  Exiting"
+    exit 1
```
Comment (Neilpang) [inline dnsapi/dns_omglol.sh:90]: don't use `exit`.   use `return 1` and check the return value in the upper level

Comment (as-kholin) [reply dnsapi/dns_omglol.sh]: Updated per comment

Comment (as-kholin) [reply dnsapi/dns_omglol.sh]: @Neilpang ready for your re-review

File: dnsapi/dns_omglol.sh [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+
+}
+
+#Usage: fulldomain txtvalue
+#Remove the txt record after validation.
+dns_omglol_rm() {
+  fulldomain=$1
+  txtvalue=$2
+  omglol_apikey="${omglol_apikey:-$(_readaccountconf_mutable omglol_apikey)}"
+  omglol_address="${omglol_address:-$(_readaccountconf_mutable omglol_address)}"
```
Comment (Neilpang) [inline dnsapi/dns_omglol.sh:61]: the name should be someting like:  OMG_ApiKey , or so.

File: dnsapi/dns_omglol.sh [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+  omg_add "$omg_address" "$authHeader" "$dnsName" "$txtvalue"
+
+}
+
+#Usage: fulldomain txtvalue
+#Remove the txt record after validation.
+dns_omglol_rm() {
+  fulldomain=$1
+  txtvalue=$2
+  omg_apikey="${omg_apikey:-$(_readaccountconf_mutable omg_apikey)}"
```
Comment (Neilpang) [inline dnsapi/dns_omglol.sh:60]: The initial should be upper case.

Comment (as-kholin) [reply dnsapi/dns_omglol.sh]: updated

**Comment by as-kholin** [comment]
Sorry for the multiple moves back to draft - found a test issue, and was correcting it (and ensuring it was corrected).

**Comment by Neilpang** [comment]
update the variable names in:
https://github.com/acmesh-official/acme.sh/wiki/dnsapi2#dns_omglol
https://github.com/acmesh-official/acme.sh/issues/5299

**Comment by as-kholin** [comment]
> update the variable names in: https://github.com/acmesh-official/acme.sh/wiki/dnsapi2#dns_omglol #5299

Updated

**Pull request merged**